frontend/src/routes/_requireAuth/data.tsx

```diff
@@ -0,0 +1,351 @@
+import { useQuery } from '@tanstack/react-query'
+import { createFileRoute } from '@tanstack/react-router'
+import { useState, useMemo } from 'react'
+
+export const Route = createFileRoute('/_requireAuth/data')({
+  component: DataDebugger,
+})
+
+interface FilterCriteria {
+  [key: string]: string
+}
+
+function DataDebugger() {
+  const { authManager } = Route.useRouteContext()
+  const [selectedLexicon, setSelectedLexicon] = useState<string>('')
+  const [isPrivate, setIsPrivate] = useState(false)
+  const [repoDid, setRepoDid] = useState<string>('')
+  const [filterText, setFilterText] = useState('')
+  const [parsedFilters, setParsedFilters] = useState<FilterCriteria>({})
+
+  // Parse filter text into key-value pairs
+  const parseFilters = (text: string): FilterCriteria => {
+    const filters: FilterCriteria = {}
+    const parts = text.trim().split(/\s+/)
+    
+    for (const part of parts) {
+      if (part.includes(':')) {
+        const [key, ...valueParts] = part.split(':')
+        const value = valueParts.join(':') // Handle values that might contain colons
+        if (key && value) {
+          filters[key] = value
+        }
+      }
+    }
+    
+    return filters
+  }
+
+  // Update filters when filter text changes
+  const handleFilterChange = (text: string) => {
+    setFilterText(text)
+    setParsedFilters(parseFilters(text))
+  }
+
+  // Fetch data based on selected lexicon, privacy setting, and optional repo DID
+  const { data, isLoading, error, refetch } = useQuery({
```

Comment:
move all this to the loader and use query params to manage state like `isPrivate`, `repoDid` etc.